packages/core/src/components/forms/numericInput.tsx

```diff
@@ -110,6 +110,9 @@ export interface INumericInputProps extends IIntentProps, IProps {
     /** The value to display in the input field. */
     value?: number | string;
 
+    /** The callback invoked when the value changes as a result of a button click. */
+    onButtonClick?(valueAsNumber: number, valueAsString: string): void;
+
     /** The callback invoked when the value changes. */
```

Comment:
add more details here to distinguish from `oBC`: "when the value changes due to typing, arrow keys, or button clicks."